### PR TITLE
sonar-scanner: Add version 4.2.0.1873

### PR DESCRIPTION
**Pull request by stevehipwell**
Added the [sonar-scanner](https://docs.sonarqube.org/latest/analysis/scan/sonarscanner/) package to compliment _SonarQube_.

### CODE REVIEW

File: bucket/sonar-scanner.json [hunk 1]
```diff
@@ -0,0 +1,29 @@
+{
+    "homepage": "https://docs.sonarqube.org/latest/analysis/scan/sonarscanner/",
+    "description": "Scan code for Sonarqube when there is no specific scanner for your build system.",
+    "license": "LGPL-3.0-only",
+    "version": "4.2.0.1873",
+    "architecture": {
+        "64bit": {
+            "url": "https://binaries.sonarsource.com/Distribution/sonar-scanner-cli/sonar-scanner-cli-4.2.0.1873.zip",
```
Comment (Ash258) [inline bucket/sonar-scanner.json]: This is not 64bit. 

WIndows 64bit url is this: https://binaries.sonarsource.com/Distribution/sonar-scanner-cli/sonar-scanner-cli-4.0.0.1744-windows.zip

anyarch version only depends on java and could be used in any environment.

Remove architecture.64bit property.

File: bucket/sonar-scanner.json [hunk 2]
```diff
@@ -0,0 +1,29 @@
...
+            "hash": "C1903A3F4CB8AD4F0EEAF8E478B667FBE9EB6EC595DB558486761404B88D0B2D"
+        }
+    },
+    "extract_dir": "sonar-scanner-4.2.0.1873",
+    "bin": [
```
Comment (Ash258) [inline bucket/sonar-scanner.json]:
```suggestion
    "bin": "bin\\sonar-scanner.bat",
```

File: bucket/sonar-scanner.json (hunk 1 above)
Comment (stevehipwell) [reply bucket/sonar-scanner.json]: Thanks for catching this @Ash258! Fixed.

File: bucket/sonar-scanner.json (hunk 2 above)
Comment (stevehipwell) [reply bucket/sonar-scanner.json]: Fixed.

File: bucket/sonar-scanner.json [hunk 3]
```diff
@@ -0,0 +1,23 @@
+{
+    "homepage": "https://docs.sonarqube.org/latest/analysis/scan/sonarscanner/",
+    "description": "Scan code for Sonarqube when there is no specific scanner for your build system.",
+    "license": "LGPL-3.0-only",
+    "version": "4.2.0.1873",
+    "url": "https://binaries.sonarsource.com/Distribution/sonar-scanner-cli/sonar-scanner-cli-4.2.0.1873.zip",
+    "hash": "C1903A3F4CB8AD4F0EEAF8E478B667FBE9EB6EC595DB558486761404B88D0B2D",
+    "extract_dir": "sonar-scanner-4.2.0.1873",
+    "bin": "bin/sonar-scanner.bat",
```
Comment (Ash258) [inline bucket/sonar-scanner.json]:
```suggestion
    "bin": "bin\\sonar-scanner.bat",
```

File: bucket/sonar-scanner.json [hunk 4]
```diff
@@ -0,0 +1,23 @@
+{
+    "homepage": "https://docs.sonarqube.org/latest/analysis/scan/sonarscanner/",
+    "description": "Scan code for Sonarqube when there is no specific scanner for your build system.",
```
Comment (Ash258) [inline bucket/sonar-scanner.json]: This description do not provide sufficient information.

This application is some sort of source code analyzer right?

Comment (stevehipwell) [reply bucket/sonar-scanner.json]: I've updated the description which will hopefully be clearer. However nobody would just download this, they would already be familiar with _SonarQube_ as you need to connect the _sonar-scanner_ to a _SonarQube_ instance (local, self hosted or SaaS).

File: bucket/sonar-scanner.json [hunk 5]
```diff
@@ -0,0 +1,24 @@
+{
+    "homepage": "https://docs.sonarqube.org/latest/analysis/scan/sonarscanner/",
+    "description": "Scan your source code to analysis it with SonarQube. Use this CLI when there is no specific scanner for your build system.",
```
Comment (Ash258) [inline bucket/sonar-scanner.json]:
```suggestion
    "description": "Source code scanner for SonarQube analyzer.",
```

**Pull request merged**